tests/test_dle.py

```diff
@@ -523,6 +523,33 @@ def test_scheduling(self):
         assert not iterations[3].is_Affine
         assert 'schedule(dynamic,chunk_size)' in iterations[3].pragmas[0].value
 
+    @pytest.mark.parametrize('so', [0, 1, 2])
+    @pytest.mark.parametrize('dim', [1, 2])
+    def test_array_reduction(self, so, dim):
+        grid = Grid(shape=(3, 3, 3))
+        d = grid.dimensions[dim]
+
+        f = Function(name='f', shape=(3,), dimensions=(d,), grid=grid, space_order=so)
```

Comment:
I would add a parameter for shape sizes and would use irregular shapes.
`nx, ny, nz = 3, 6, 3`
f= ......shape=(shape[dim])..

Reply:
Not sure why the size would make any difference.